mlflow/entities/_job.py

```diff
@@ -0,0 +1,74 @@
+from mlflow.entities._mlflow_object import _MlflowObject
+from mlflow.entities._job_status import JobStatus
+
+
+class Job(_MlflowObject):
+    """
+    MLflow entity representing a Job.
+    """
+
+    def __init__(
+        self,
+        job_id: str,
+        creation_time: int,
+        function: str,
+        params: str,
+        status: JobStatus,
+        result: str | None,
+        retry_count: int = 0,
+    ):
+        super().__init__()
+        self._job_id = job_id
+        self._creation_time = creation_time
+        self._function = function
+        self._params = params
+        self._status = status
+        self._result = result
+        self._retry_count = retry_count
+
+    @property
+    def job_id(self) -> str:
+        """String containing job ID."""
+        return self._job_id
+
+    @property
+    def creation_time(self) -> int:
+        """Creation timestamp of the job, in number of milliseconds since the UNIX epoch."""
+        return self._creation_time
+
+    @property
+    def function(self) -> str:
+        """
+        String containing the function full name, in the form of `<module_name>.<function_name>`
+        """
+        return self._function
```

Comment:
What happens if we upgrade MLflow and rename the function that gets executed? Existing jobs in the database will reference and invalid function name. Thoughts on how we can avoid this?

Reply:
We should keep the function `<module_name>.<function_name>` unchanged, otherwise the unfinished jobs can't be resumed when MLflow server restarts (with new Mlflow version installed)

`@developer_stable` sounds good.